api/src/main/java/org/pmiops/workbench/institution/InstitutionEmailDomainMapper.java

```diff
@@ -0,0 +1,35 @@
+package org.pmiops.workbench.institution;
+
+import java.util.Collections;
+import java.util.Optional;
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.stream.Collectors;
+import org.mapstruct.Mapper;
+import org.mapstruct.NullValueMappingStrategy;
+import org.pmiops.workbench.db.model.DbInstitution;
+import org.pmiops.workbench.db.model.DbInstitutionEmailDomain;
+import org.pmiops.workbench.model.Institution;
+import org.pmiops.workbench.utils.mappers.MapStructConfig;
+
+@Mapper(
+    config = MapStructConfig.class,
+    nullValueMappingStrategy = NullValueMappingStrategy.RETURN_DEFAULT)
+public interface InstitutionEmailDomainMapper {
+  default Set<DbInstitutionEmailDomain> modelToDb(
+      final Institution modelInstitution, final DbInstitution dbInstitution) {
+    return Optional.ofNullable(modelInstitution.getEmailDomains()).orElse(Collections.emptyList())
+        .stream()
+        .distinct()
+        .map(
+            domain ->
+                new DbInstitutionEmailDomain().setEmailDomain(domain).setInstitution(dbInstitution))
```

Comment:
If you define a mapping from domain and dbInst to DbInstitutionEmailDomain, MapStruct should be able to write this code I think.

Reply:
updated